src/fields/fq.rs

```diff
@@ -769,6 +769,21 @@ impl FieldExt for Fq {
     }
 }
 
+#[cfg(feature = "gpu")]
+impl ec_gpu::GpuField for Fq {
+    fn one() -> alloc::vec::Vec<u32> {
+        crate::fields::u64_to_u32(&R.0[..])
+    }
+
+    fn r2() -> alloc::vec::Vec<u32> {
+        crate::fields::u64_to_u32(&R2.0[..])
```

Comment:
Assuming my analysis in my previous comment is correct:
```suggestion
        crate::fields::u64_to_u32(&R3.0[..])
```